stale_read_user_guide.md

```diff
@@ -0,0 +1,186 @@
+
+```
+Region read progress:
+    exist: true,
+    safe_ts: 0,
+    applied_index: 92,
+    pending front item (oldest) ts: 0,
+    pending front item (oldest) applied index: 0,
+    pending back item (latest) ts: 0,
+    pending back item (latest) applied index: 0,
+    paused: false,
+Resolver:
+    exist: true,
+    resolved_ts: 0,
+    tracked index: 92,
+    number of locks: 0,
+    number of transactions: 0,
+    stopped: false,
+```
+
+This information can help determine:
+
+- Whether locks are blocking resolved-ts.
+- Whether the apply index is too small, preventing safe-ts from being updated.
+- If this is a follower peer, whether the leader is sending a sufficiently new resolved-ts.
+
+## Troubleshooting Tips
+
+### Handling Slowly Committing Transactions
+
+Slowly committing transactions are often large transactions. Their prewrite phases leave some locks, but it takes too long before commit phases clean the locks. Try to identify the transactions to which the locks belong and determine why they exist (through logs, etc.)
+
+Useful actions you can take
+
+- Specify `--log` option in the tikv-ctl command, and check TiKV logs to find the specific locks with their start_ts.
+- Search the start_ts in both TiDB and TiKV logs, see if anything is wrong with this transaction
+    - If a query exceeds 60 seconds, an `Expensive query`  log will be printed. The log includes the SQL statement. You can use start_ts to match the log.
+    - `[2023/07/17 19:32:09.403 +08:00] [WARN] [expensivequery.go:145] [expensive_query] [cost_time=60.025022732s] [cop_time=0.00346666s] [process_time=8.358409508s] [wait_time=0.013582596s] [request_count=278] [total_keys=9943616] [process_keys=9943360] [num_cop_tasks=278] [process_avg_time=0.030066221s] [process_p90_time=0.045296042s] [process_max_time=0.052828934s] [process_max_addr=192.168.31.244:20160] [wait_avg_time=0.000048858s] [wait_p90_time=0.00006057s] [wait_max_time=0.00040991s] [wait_max_addr=192.168.31.244:20160] [stats=t:442916666913587201] [conn=2826881778407440457] [user=root] [database=test] [table_ids="[100]"] [`**`txn_start_ts`**`=442916790435840001] [mem_max="2514229289 Bytes (2.34 GB)"] [sql="update t set b = b + 1"]`
+- [CLUSTER_TIDB_TRX](https://docs.pingcap.com/tidb/dev/information-schema-tidb-trx#cluster_tidb_trx) table could be useful to find active transactions if logs can't tell us enough info about the locks.
+- `show processlist` shows current active connections and their time spent on current command, but it doesn't show start_ts.
+
+If the locks exist due to ongoing large transactions, you should consider modifying your application logic as these locks can hinder the progress of resolve-ts.
+
+If the locks do not belong to any ongoing transactions, it might be due to a coordinator (TiDB) crashing after it prewrites the lock.. In this case, TiDB should automatically resolve the locks. No action is required unless you find that the problem persists and cannot recover.
```

Comment:
````suggestion
The following list some actions you can take:

- Specify `--log` option in the `tikv-ctl` command and check TiKV logs to find the specific locks with their start_ts.
- Search the start_ts in both TiDB and TiKV logs to identify issues with the transaction.

    If a query takes over 60 seconds, an `expensive_query` log is printed with the SQL statement. You can use the start_ts value to match the log. The following is an example:

    ```log
    [2023/07/17 19:32:09.403 +08:00] [WARN] [expensivequery.go:145] [expensive_query] [cost_time=60.025022732s] [cop_time=0.00346666s] [process_time=8.358409508s] [wait_time=0.013582596s] [request_count=278] [total_keys=9943616] [process_keys=9943360] [num_cop_tasks=278] [process_avg_time=0.030066221s] [process_p90_time=0.045296042s] [process_max_time=0.052828934s] [process_max_addr=192.168.31.244:20160] [wait_avg_time=0.000048858s] [wait_p90_time=0.00006057s] [wait_max_time=0.00040991s] [wait_max_addr=192.168.31.244:20160] [stats=t:442916666913587201] [conn=2826881778407440457] [user=root] [database=test] [table_ids="[100]"] [**txn_start_ts**=442916790435840001] [mem_max="2514229289 Bytes (2.34 GB)"] [sql="update t set b = b + 1"]
    ```

- Use the [`CLUSTER_TIDB_TRX`](/information-schema/information-schema-tidb-trx.md#cluster_tidb_trx) table to find active transactions if you cannot get enough information about the locks from logs.
- Execute [`SHOW PROCESSLIST`](/sql-statements/sql-statement-show-processlist.md) to view the current sessions connected to the same TiDB server and their time spent on current statement. But it does not show start_ts.

If the locks exist due to ongoing large transactions, consider modifying your application logic as these locks can hinder the progress of resolve-ts.

If the locks do not belong to any ongoing transactions, it might be due to a coordinator (TiDB) crashing after it prewrites the locks. In this case, TiDB will automatically resolve the locks. No action is required unless the problem persists.
````